XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs

```diff
@@ -60,17 +60,19 @@ public string PatchPath
             get; private set;
         }
 
-        public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string outputFileName)
-            : this(patchName, uri, outputFileName, InvisibleMessages.XEN_UPDATE)
-        { }
+        private bool DownloadedUpdateFromUri

```

Comment:
I suggest renaming this to downloadUpdate and change it to a private readonly variable which is set once in the constructor (similar to address, zippedFileName, etc). There is no need to have it as a property with a getter if it never changes.